game-core/src/main/java/games/strategy/triplea/attachments/AbstractUserActionAttachment.java

```diff
@@ -71,29 +76,60 @@ private void resetText() {
     m_text = "";
   }
 
-  /**
-   * @param s
-   *        the amount you need to pay to perform the action.
-   */
+  @Deprecated
   private void setCostPu(final String s) {
-    m_costPU = getInt(s);
+    setCostPu(getInt(s));
   }
 
+  @Deprecated
   public void setCostPu(final Integer s) {
-    m_costPU = s;
+    final Resource r = getData().getResourceList().getResource(Constants.PUS);
+    m_costResources.put(r, s);
   }
 
-  /**
-   * @return The amount you need to pay to perform the action.
-   */
+  @Deprecated
   public int getCostPu() {
     return m_costPU;
   }
 
+  @Deprecated
   private void resetCostPu() {
     m_costPU = 0;
   }
 
+  private void setCostResources(final String value) throws GameParseException {
+    final String[] s = value.split(":");
+    if (s.length <= 0 || s.length > 2) {
+      throw new GameParseException(
+          "costResources cannot be empty or have more than two fields" + thisErrorMsg());
```

Comment:
should/could the parameter be logged in this exception? (Same for any other exception in this method)

Reply:
Yeah, I'll add the value to it.